### DISCUSSION TITLE
reason for CCF memory usage

### DISCUSSION THREAD

**Discussion by jochen-schuettler** [cloud-carbon-footprint/cloud-carbon-footprint#1335] [Q&A]
We've installed CCF with Terraform on AWS EC2. It takes up more than 4 GB in memory. Why is that so? What need is there for so much memory?

**Comment by 4upz** [comment]
@jochen-schuettler Admittedly, I haven't been close to our Terraform template but I can provide some thoughts I have after giving it a quick look:

- The terraform install script installs CCF by cloning the repository and then running the app in dev mode. This could be a partial cause of the memory issue as CCF is running in an unoptimized dev environment using ts-node to locally compile all of the CCF packages that the API relies on. We could improve this by updating the `install.sh` file to install a production build of ccf before running it.
- Another reason could be the size of your data in a single request. CCF makes the query for usage data and holds it in memory to calculate it before saving it to the DB or local cache file. If you are making a request for a large date range with granular grouping (i.e. day) then this could cause a spike in memory.

My primary bet is on the latter, as I have managed to run a CCF instance on a t2.micro for a personal account. If running CCF using accounts with a large usage, that could be the primary reason for the additional memory where it retains those resources after handling a large request. A quick win on improving this would be making the change to the terraform file as highlighted in the first point.

**Reply by 4upz** [reply]
@olivercodes Would love your thoughts on this as well and whether you have tried the terraform route or have encountered this before
